hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java

```diff
@@ -1127,7 +1128,9 @@ private boolean isTableInGroup(TableName tableName, String groupName,
           .isTableState(tn, TableState.State.DISABLED, TableState.State.DISABLING)) {
           continue;
         }
-        if (region.isSplitParent()) {
+        if (
+          regionStates.getOrCreateRegionStateNode(region).getState().equals(RegionState.State.SPLIT)
```

Comment:
getOrCreateRegionStateNode could have a side-effect when there is no RegionStateNode for 'region'. In this case, we'll add to RegionState a new RSN for 'region' if none present.... Do we want to do this? 

You make the above call because you don't have access to the internal createRegionStateNode(regionInfo ?